### PR TITLE
Apply the c-code template

### LINKED ISSUE #33: Add support to release aarch64 wheels
Problem
---------
On aarch64, pip install zope-index builds the wheels from source code and then install it. It requires user to have development environment installed on his system. also, it take some time to build the wheels than downloading and extracting the wheels from pypi.

Resolution
-----------
On aarch64, pip install zope-index should download the wheels from pypi

Please let me know your interest on releasing aarch64 wheels. I can help in this. Are there any plans for moving to github actions?

### PR DESCRIPTION
**Pull request by mgedmin**
We want to switch to GitHub Actions, enable aarch64 wheels, and generally modernize the packaging (e.g. add support for Python 3.9).

I've tried to run the config tool with `-t c-code --no-commit --no-flake8 --with-pypy --with-legacy-python` and looked at git diff.  Here's a checklist of things that might need tweaking:

- [x] `.gitignore` loses `*.dll`, which might be inconvenient if we ever have any developers doing any development on Windows?  but also we don't list `*.so` either, so maybe it wasn't necessary?
- [x] `MANIFEST.in` loses `include *.sh`, which is something that IMHO the template itself should take care about, given that it generates .manylinux*.sh files!
- [x] the config tool ought to notice the presence of `appveyor.yml` and automatically turn on `--with-appveyor` → https://github.com/zopefoundation/meta/pull/113
- [x] the config tool ought to notice the presence of `docs/conf.py` and automatically turn on `--with-docs` → https://github.com/zopefoundation/meta/pull/113
- [x] maybe we should add `recursive-include src *.rst` to MANIFEST.in by default?  Or at least if any such files exist?
- [x] check-manifest complains about `docs/make.bat`, do I `git rm` that file?  IIRC that `recursive-include docs *.bat` was intentionally removed from the MANIFEST.in in the template for some reason
- [x] does zope.index support PURE_PYTHON, or are we doubling the number of test environments unnecessarily?  [Yes](https://github.com/zopefoundation/zope.index/blob/ad6a2fb5f39310f5c062b8d2cc6ea52a49d785db/src/zope/index/text/okapiindex.py#L200-L206).
- [x] do we need `docs` in [testenv] extras?  I think not.
- [x] `--fail-under` was 100, now it drops to 0, I wish the metaconfig tool could autodetect this value too.  It probably can't: we now enable branch coverage, so the actual value drops to 99.16%
- [x] `tox -p auto` doesn't show the results of the `coverage` environment; should it?  maybe not, it's very long.
- [x] I need to generate a PyPI token and configure the TWINE_PASSWORD secret variable in GitHub settings for this repo
- [x] heyy where did the `python setup.py bdist_wheel` disappeared from appveyor.yml??   I think it's a bug in the template: if it has the appveyor upload, it should also have a bdist_wheel.

Closes #33.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -29,20 +27,22 @@ install:
...
+  - pip install -U setuptools wheel
+  - pip install -U -e .[test]
+
+matrix:
+  fast_finish: true
 
-build_script:
-  - pip install wheel
-  - python -W ignore setup.py -q bdist_wheel
+build: false
```
Comment (mgedmin) [inline appveyor.yml:37]: Um, what???

If we're not building Windows wheels, why do we have `twine upload`?

File: appveyor.yml [hunk 2]
```diff
@@ -1,11 +1,6 @@
-# Currently the builds use @mgedmin's Appveyor account.  The PyPI token belongs
-# to zope.wheelbuilder, which is managed by @mgedmin and @dataflake.
-
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/c-code
 environment:
-  global:
-    TWINE_USERNAME: __token__
-    TWINE_PASSWORD:
-      secure: aoZC/+rvJKg8B5GMGIxd1VTV3PcyYlv24IVlfMnG7u6WiuD8XHkwbNCE9e6o7me9uvONYgQBwedB0w3pEOjSYCZnGLHD8wPpSyvU6mJ/sm5zefc/k4/MwNE1QHlT4+e+W7YnjXZCZYCTihoGVxCaCOpttBrqt2ccNpXlfm9rJM1/AXL5e5TvahSnRtV4HfEt3BEtbO3GGrqXVa6wk+T4f2ywIKMmZ+xmav9eslRCnI6pX8dWs6Ulw8fr5QUycZY+ARTXyPYz4iOrNSVzs4uIQg==
```
Comment (mgedmin) [inline appveyor.yml:8]: Oops, something that shouldn't have been lost.

**Comment by mgedmin** [comment]
I think this is ready.  Please review!

File: .gitignore [hunk 3]
```diff
@@ -1,15 +1,30 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/c-code
+*.egg-info/
+*.profraw
 *.pyc
+*.pyo
 *.so
-*.dll
```
[the remark this answers is not in the document]
Comment (icemac) [reply .gitignore:3]: I added `*.dll` to to `meta/config/default/gitignore`.

File: .meta.toml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "# Currently the builds use @mgedmin's Appveyor account.  The PyPI token",
+    "# belongs to zope.wheelbuilder, which is managed by @mgedmin and @dataflake.",
+    "",
+    "global:",
+    "  TWINE_USERNAME: __token__",
+    "  TWINE_PASSWORD:",
+    "    secure: aoZC/+rvJKg8B5GMGIxd1VTV3PcyYlv24IVlfMnG7u6WiuD8XHkwbNCE9e6o7me9uvONYgQBwedB0w3pEOjSYCZnGLHD8wPpSyvU6mJ/sm5zefc/k4/MwNE1QHlT4+e+W7YnjXZCZYCTihoGVxCaCOpttBrqt2ccNpXlfm9rJM1/AXL5e5TvahSnRtV4HfEt3BEtbO3GGrqXVa6wk+T4f2ywIKMmZ+xmav9eslRCnI6pX8dWs6Ulw8fr5QUycZY+ARTXyPYz4iOrNSVzs4uIQg==",
+    ]
+build-script = [
+    "- python -W ignore setup.py -q bdist_wheel",
```
Comment (icemac) [inline .meta.toml:39]: This is now automatically set for the `c-code` type in `meta/config`.

File: .meta.toml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+[tox]
+use-flake8 = false
+
+[coverage]
+fail-under = 99
+
+[manifest]
+additional-rules = [
+    "include *.sh",
```
Comment (icemac) [inline .meta.toml:23]: To automatically add this I created https://github.com/zopefoundation/meta/pull/117.

**Review by icemac — APPROVED** [review]
LGTM.

**Comment by mgedmin** [comment]
Thank you!

**Pull request merged**